Compentio.SourceMapper.Generator/Processors/Result.cs

```diff
@@ -76,7 +76,7 @@ internal static IResult Error(Exception exception)
 
         public IEnumerable<DiagnosticsInfo>? Diagnostics => _diagnostics;
 
-        public bool IsSuccess => !_diagnostics.Any(info => info.Exception is not null);
+        public bool IsSuccess => _diagnostics is null || !_diagnostics.Any(info => info.Exception is not null);
```

Comment:
To check:
```cs
_diagnostics is null || !_diagnostics.Any(info => info.Exception is not null)
```
vs

```cs
!_diagnostics?.Any(info => info.Exception is not null)
```
when `_diagnostics is null`

Reply:
.Any throw exception if argument is null. When we return Result.Ok(string) from method, and we would like to check IsSuccess, we got error due to null of _diagnostic object on this moment, and so info.Exception throw exception.
I think my changes schould stay, but I will test this via unit tests.

Reply:
Please, check `!_diagnostics**?**`

Reply:
The ? mark lead to change in IsSuccess type to bool?, and in consequence, we need to change all references to this field. Is should stay as now.